apiserver/admin_test.go

```diff
@@ -527,7 +527,7 @@ func (s *loginSuite) TestLoginValidationDuringUpgrade(c *gc.C) {
 		c.Assert(err, jc.ErrorIsNil)
 
 		err = st.APICall("Client", 1, "", "DestroyModel", nil, nil)
-		c.Assert(err, gc.ErrorMatches, ".*upgrade in progress - Juju functionality is limited.*")
+		c.Assert(errors.Cause(err), gc.DeepEquals, &rpc.RequestError{Message: "upgrade in progress", Code: ""})
```

Comment:
This still isn't right, the Code should be "upgrade in progress", not the message. But at least it's coming in an rpc.RequestError.